pkg/engine/availabilitysets.go

```diff
@@ -11,6 +11,13 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
+// PlatformFaultDomainCountUnset indicates an ARM expression should replace this value.
+const PlatformFaultDomainCountNotSet = 0
+
+// PlatformFaultDomainCount is the global value used to populate the fault domain count
+// for a VMAS. It may be changed when upgrade inspects an existing cluster.
+var PlatformFaultDomainCount int32 = PlatformFaultDomainCountNotSet
```

Comment:
Can we just add an additional `PlatformFaultDomainCount *int` property to the `AgentPoolProfile` type? Then we can infer "not set" by a `nil` value.

Reply:
Done.